### PR TITLE
fix(patch): [sc-3782] DistributedSystem.connectToService() interrupts if a running task is cancelled.

### PR DESCRIPTION
**Pull request by ser-0xff**
## Description

DistributedSystem.connectToService() now will be interrupted if the running task is cancelled.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. 

## Minimal checklist:

- [X] I have performed a self-review of my own code 
- [ ] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [X] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 1]
```diff
@@ -111,19 +119,25 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
...
+        var cancelled: Bool = false
 
-        init(_ serviceName: String, _ id: UInt64, _ actorSystem: DistributedSystem) {
-            self.serviceName = serviceName
-            self.id = id
+        init(_ actorSystem: DistributedSystem) {
             self.actorSystem = actorSystem
         }
 
         public func cancel() -> Bool {
```
Comment (dimlio) [inline Sources/DistributedSystem/DistributedSystem.swift:131]: Make result discardable?

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 2]
```diff
@@ -386,58 +398,65 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
 
         var cancellationToken = cancellationToken
```
Comment (dimlio) [inline Sources/DistributedSystem/DistributedSystem.swift:399]:
```suggestion
        let cancellationToken = cancellationToken ?? makeCancellationToken()
```
?

**Review by dimlio — APPROVED** [review]

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 3]
```diff
@@ -111,19 +119,25 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
         }
     }
 
-    public struct CancellationToken {
-        let serviceName: String
-        let id: UInt64
-        let actorSystem: DistributedSystem
+    public class CancellationToken: Hashable {
```
[the remark this answers is not in the document]
Comment (ser-0xff) [reply Sources/DistributedSystem/DistributedSystem.swift:122]: Fixed.

File: Sources/DistributedSystem/DistributedSystem.swift (hunk 1 above)
Comment (ser-0xff) [reply Sources/DistributedSystem/DistributedSystem.swift:131]: Not sure it is required.

File: Sources/DistributedSystem/DistributedSystem.swift (hunk 2 above)
Comment (ser-0xff) [reply Sources/DistributedSystem/DistributedSystem.swift:399]: Fixed.

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 4]
```diff
@@ -111,19 +119,25 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
...
-            actorSystem.cancel(serviceName, id)
+            actorSystem.cancel(self)
+        }
+
+        public func hash(into hasher: inout Hasher) {
+            hasher.combine(Unmanaged.passUnretained(self).toOpaque())
+        }
+
+        public static func == (lhs: DistributedSystem.CancellationToken, rhs: DistributedSystem.CancellationToken) -> Bool {
+            return (lhs === rhs)
```
[the remark this answers is not in the document]
Comment (ser-0xff) [reply Sources/DistributedSystem/DistributedSystem.swift:140]: Fixed.

**Pull request merged**